src/vs/base/common/oauth.ts

```diff
@@ -866,3 +866,21 @@ export function getResourceServerBaseUrlFromDiscoveryUrl(discoveryUrl: string):
 
 	return baseUrl.toString();
 }
+
+/**
+ * Normalizes a URL for comparison purposes by ensuring consistent formatting.
+ * Handles differences in hostname case, trailing slashes for root paths, etc.
+ *
+ * @param url The URL to normalize
+ * @returns The normalized URL string
+ */
+export function normalizeUrlForComparison(url: string): string {
+	const parsed = new URL(url);
+	
+	// Ensure root paths have a trailing slash
+	if (!parsed.pathname || parsed.pathname === '') {
```

Comment:
Do you need to do this at all? I thought using URL would normalize the url already?